src/client/src/Domains/Accounts/SignUpForm/index.tsx

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import Typography from '@material-ui/core/Typography';
+import Grid from '@material-ui/core/Grid';
+import Paper from '@material-ui/core/Paper';
+import { Formik, Form, Field } from 'formik';
+import * as yup from 'yup';
+
+import useApi from 'hooks/useApi';
+import useSnack from 'hooks/useSnack';
+import Loader from 'Components/Loader';
+import { TextFormField } from 'Components/TextFormField';
+import { SelectFormField } from 'Components/SelectFormField';
+import Button from 'Components/Button';
+import {
+    signUp,
+    roles,
+} from 'Domains/Accounts/api/api';
+
+export interface ISignUpForm {
+    email: string;
+    password: string;
+    confirmedPassword: string;
+    firstName: string;
+    lastName: string;
+    role: string;
+}
+
+const formInitialValues: ISignUpForm = {
+    email: '',
+    password: '',
+    confirmedPassword: '',
+    firstName: '',
+    lastName: '',
+    role: '',
+};
+
+const formSchema = yup.object({email: yup.string().required().email('Please enter valid emai.'),
+    password: yup.string().required('Password is required.').matches(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,}$/,
+        "Password must contain minimum 8 characters, at least one lowercase alphabet, one uppercase alphabet, and one number."),
+    confirmedPassword: yup.mixed().required('Please confirm password.').test('match', 'Passwords do not match.', function (pass) {
```

Comment:
instead of using a function here ( "function(pass)" ) can we an arrow function like this  () > { ... } and don't pass this pass parameter since it is not used anywhere in the function body.

Reply:
It didn't work. When I do this, I can't use this.parent.password and this.parent confirmedPassword inside the function anymore.